aggregators/aggregator_test.go

```diff
@@ -1088,7 +1088,8 @@ func BenchmarkAggregateCombinedMetrics(b *testing.B) {
 	}
 }
 
-func BenchmarkAggregateBatch(b *testing.B) {
+func BenchmarkAggregateBatchSerial(b *testing.B) {
```

Comment:
Can this be achieved by running BenchmarkAggregateBatchParallel with `-cpu=1`?

Reply:
I believe there is a subtle difference to it. Specifically, if the AggregateBatch function is implemented using multiple goroutines, passing `-cpu=1` will set GOMAXPROCS and will affect the performance of AggregateBatch.

An example of this is 
```
package playground

import (
	"fmt"
	"runtime"
	"sync"
	"testing"
)

func factorial(x int64) int64 {
	if x <= 1 {
		return 1
	}
	return factorial(x-1) * factorial(x-2)
}

func BenchmarkB(b *testing.B) {
	fmt.Println(runtime.GOMAXPROCS(-1))
	b.ReportAllocs()
	for i := 0; i < b.N; i++ {
		N := 10
		wg := sync.WaitGroup{}
		for j := 0; j < N; j++ {
			wg.Add(1)
			go func() {
				factorial(30)
			}()
			wg.Done()
		}
		wg.Wait()
	}
}
```

When `-cpu=1` is passed, it has 100% CPU usage on my machine (saturates 1 thread). But if `-cpu=16` is passed, it has 1600% CPU usage (saturates 16 threads). Therefore, `-cpu` doesn't only affect the concurrency of test but also how much cpu is available to the function being benched.

Of course, in practice it may not make any difference in this case.

Reply:
Fair point. I'm not sure that we should cater for AggregateBatch ever internally using multiple goroutines, but I'm not too fussed - just thought it might make analysis simpler.